equtrn/doxygen/man/man3/shotfile2dg.3
.TH "shotfile2dg" 3 "21 Jul 2008" "Version 2008-06" "DivGeo/dg/equtrn" \" -*- nroff -*-
.ad l
.nh
.SH NAME
shotfile2dg \- Translation of the AUG shotfile equilibrium data into dg-compatible format.  

.PP
.SH SYNOPSIS
.br
.PP
.SS "Variables"

.in +1c
.ti -1c
.RI "real *8, dimension(ngpr) \fBgpr\fP"
.br
.ti -1c
.RI "real *8, dimension(ngpz) \fBgpz\fP"
.br
.ti -1c
.RI "real *8, dimension(ngpr, ngpz) \fBpfmd\fP"
.br
.ti -1c
.RI "real *8 \fBrcntc\fP"
.br
.ti -1c
.RI "real *8 \fBbtorc\fP"
.br
.ti -1c
.RI "real *8 \fBpsilim\fP"
.br
.ti -1c
.RI "character *256 \fBarg\fP"
.br
.ti -1c
.RI "character *256 \fBfilename\fP"
.br
.ti -1c
.RI "character \fBexpnam\fP"
.br
.ti -1c
.RI "character \fBdianam\fP"
.br
.ti -1c
.RI "integer \fBnshot\fP"
.br
.ti -1c
.RI "integer \fBnedit\fP"
.br
.ti -1c
.RI "integer \fBiarg\fP"
.br
.ti -1c
.RI "integer \fBiargc\fP"
.br
.ti -1c
.RI "integer \fBlnblnk\fP"
.br
.ti -1c
.RI "real \fBtshot\fP"
.br
.ti -1c
.RI "integer \fBmdim\fP"
.br
.ti -1c
.RI "integer \fBlpfxdim\fP"
.br
.ti -1c
.RI "integer \fBlin\fP"
.br
.ti -1c
.RI "real, dimension(0:\fBmdim\fP) \fBri\fP"
.br
.ti -1c
.RI "real, dimension(0:\fBmdim\fP) \fBzj\fP"
.br
.ti -1c
.RI "real, dimension(0:\fBmdim\fP, 0:\fBmdim\fP) \fBpfm\fP"
.br
.ti -1c
.RI "real, dimension(0:\fBlpfxdim\fP) \fBpfxx\fP"
.br
.ti -1c
.RI "real, dimension(0:\fBlpfxdim\fP) \fBrpfx\fP"
.br
.ti -1c
.RI "real, dimension(0:\fBlpfxdim\fP) \fBzpfx\fP"
.br
.ti -1c
.RI "real, dimension(\fBlin\fP) \fBrin\fP"
.br
.ti -1c
.RI "real, dimension(\fBlin\fP) \fBzin\fP"
.br
.ti -1c
.RI "real, dimension(\fBlin\fP) \fBbr\fP"
.br
.ti -1c
.RI "real, dimension(\fBlin\fP) \fBbz\fP"
.br
.ti -1c
.RI "real, dimension(\fBlin\fP) \fBbt\fP"
.br
.ti -1c
.RI "real, dimension(\fBlin\fP) \fBfpf\fP"
.br
.ti -1c
.RI "real, dimension(\fBlin\fP) \fBfjp\fP"
.br
.in -1c
.SH "Detailed Description"
.PP 
Translation of the AUG shotfile equilibrium data into dg-compatible format. 

1st arg == shot number
.PP
2nd arg == time
.PP
[3rd arg == diagnostic (EQI)]
.PP
[4th arg == experiment (AUGD)]
.PP
[5th arg == edition (0)]
.PP
The output filename is derived from the shot, time, experiment, diagnostic and edition
.PP
Specific AUG library routines are used to access the data
.PP
\fBVersion:\fP
.RS 4
16.11.95 20:04 
.RE
.PP

.SH "Variable Documentation"
.PP 
.SS "character*256 \fBshotfile2dg::arg\fP"
.PP
Definition at line 31 of file shotfile2dg.f.
.SS "real,dimension(\fBlin\fP) \fBshotfile2dg::br\fP"
.PP
Definition at line 37 of file shotfile2dg.f.
.SS "real,dimension(\fBlin\fP) \fBshotfile2dg::bt\fP"
.PP
Definition at line 37 of file shotfile2dg.f.
.SS "real*8 \fBshotfile2dg::btorc\fP"
.PP
Definition at line 30 of file shotfile2dg.f.
.SS "real,dimension(\fBlin\fP) \fBshotfile2dg::bz\fP"
.PP
Definition at line 37 of file shotfile2dg.f.
.SS "character \fBshotfile2dg::dianam\fP"
.PP
Definition at line 32 of file shotfile2dg.f.
.SS "character \fBshotfile2dg::expnam\fP"
.PP
Definition at line 32 of file shotfile2dg.f.
.SS "character*256 \fBshotfile2dg::filename\fP"
.PP
Definition at line 31 of file shotfile2dg.f.
.SS "real,dimension(\fBlin\fP) \fBshotfile2dg::fjp\fP"
.PP
Definition at line 37 of file shotfile2dg.f.
.SS "real,dimension(\fBlin\fP) \fBshotfile2dg::fpf\fP"
.PP
Definition at line 37 of file shotfile2dg.f.
.SS "real*8,dimension(ngpr) \fBshotfile2dg::gpr\fP"
.PP
Definition at line 28 of file shotfile2dg.f.
.SS "real*8,dimension(ngpz) \fBshotfile2dg::gpz\fP"
.PP
Definition at line 28 of file shotfile2dg.f.
.SS "integer \fBshotfile2dg::iarg\fP"
.PP
Definition at line 33 of file shotfile2dg.f.
.SS "integer \fBshotfile2dg::iargc\fP"
.PP
Definition at line 33 of file shotfile2dg.f.
.SS "integer \fBshotfile2dg::lin\fP"
.PP
Definition at line 35 of file shotfile2dg.f.
.SS "integer \fBshotfile2dg::lnblnk\fP"
.PP
Definition at line 33 of file shotfile2dg.f.
.SS "integer \fBshotfile2dg::lpfxdim\fP"
.PP
Definition at line 35 of file shotfile2dg.f.
.SS "integer \fBshotfile2dg::mdim\fP"
.PP
Definition at line 35 of file shotfile2dg.f.
.SS "integer \fBshotfile2dg::nedit\fP"
.PP
Definition at line 33 of file shotfile2dg.f.
.SS "integer \fBshotfile2dg::nshot\fP"
.PP
Definition at line 33 of file shotfile2dg.f.
.SS "real,dimension(0:\fBmdim\fP,0:\fBmdim\fP) \fBshotfile2dg::pfm\fP"
.PP
Definition at line 37 of file shotfile2dg.f.
.SS "real*8,dimension(ngpr,ngpz) \fBshotfile2dg::pfmd\fP"
.PP
Definition at line 29 of file shotfile2dg.f.
.SS "real,dimension(0:\fBlpfxdim\fP) \fBshotfile2dg::pfxx\fP"
.PP
Definition at line 37 of file shotfile2dg.f.
.SS "real*8 \fBshotfile2dg::psilim\fP"
.PP
Definition at line 30 of file shotfile2dg.f.
.SS "real*8 \fBshotfile2dg::rcntc\fP"
.PP
Definition at line 30 of file shotfile2dg.f.
.SS "real,dimension(0:\fBmdim\fP) \fBshotfile2dg::ri\fP"
.PP
Definition at line 37 of file shotfile2dg.f.
.SS "real,dimension(\fBlin\fP) \fBshotfile2dg::rin\fP"
.PP
Definition at line 37 of file shotfile2dg.f.
.SS "real,dimension(0:\fBlpfxdim\fP) \fBshotfile2dg::rpfx\fP"
.PP
Definition at line 37 of file shotfile2dg.f.
.SS "real \fBshotfile2dg::tshot\fP"
.PP
Definition at line 34 of file shotfile2dg.f.
.SS "real,dimension(\fBlin\fP) \fBshotfile2dg::zin\fP"
.PP
Definition at line 37 of file shotfile2dg.f.
.SS "real,dimension(0:\fBmdim\fP) \fBshotfile2dg::zj\fP"
.PP
Definition at line 37 of file shotfile2dg.f.
.SS "real,dimension(0:\fBlpfxdim\fP) \fBshotfile2dg::zpfx\fP"
.PP
Definition at line 37 of file shotfile2dg.f.
.SH "Author"
.PP 
Generated automatically by Doxygen for DivGeo/dg/equtrn from the source code.


equtrn/src/shotfile2dg.f
!> Translation of the AUG shotfile equilibrium data into dg-compatible format
!>
!> 1st arg == shot number
!>
!> 2nd arg == time
!>
!> [3rd arg == diagnostic (EQI)]
!>
!> [4th arg == experiment (AUGD)]
!>
!> [5th arg == edition (0)]
!>
!> The output filename is derived from the shot, time, experiment, diagnostic and edition
!>
!> Specific AUG library routines are used to access the data
!>
!> \version 16.11.95 20:04

      program shotfile2dg
c
c  version : 16.11.95 20:04
c
c=====================================================
c*** Translation of shotfile equilibrium data into dg-compatible format
c=====================================================
      parameter (pi=3.14159 26535 89793)
      parameter (ngpr=257, ngpz=257, ngpf=257)
      real*8 gpr(ngpr),gpz(ngpz)
      real*8 pfmd(ngpr,ngpz)
      real*8 rcntc,btorc,psilim
      character*256 arg,filename
      character expnam*4, dianam*3
      integer nshot,nedit,iarg,iargc,lnblnk
      real tshot
      integer mdim,lpfxdim,lin
      parameter(mdim=256,lpfxdim=4,lin=1)
      real ri(0:mdim),zj(0:mdim),pfm(0:mdim,0:mdim),
     1 pfxx(0:lpfxdim),rpfx(0:lpfxdim),zpfx(0:lpfxdim),
     2 rin(lin),zin(lin),br(lin),bz(lin),bt(lin),fpf(lin),fjp(lin)
c=====================================================
c
      dianam='EQI'
      expnam='AUGD'
      m=mdim
      n=mdim
      nedit=0
      iarg=iargc()
      write(*,*) 'iarg',iarg
      if(iarg.ge.2) then
	call getarg(1,arg)
	read(arg,*) nshot
	call getarg(2,arg)
	read(arg,*) tshot
	if(iarg.ge.3) call getarg(3,dianam)
	if(iarg.ge.4) call getarg(4,expnam)
	if(iarg.ge.5) then
	  call getarg(5,arg)
	  read(arg,*) nedit
	endif
      else
	write(*,*) '1st arg == shot number'
	write(*,*) '2nd arg == time'
	write(*,*) '[3rd arg == diagnostic (EQI)]'
	write(*,*) '[4th arg == experiment (AUGD)]'
	write(*,*) '[5th arg == edition (0)]'
	stop
      endif
      write(*,*) expnam,dianam,nshot,tshot
      write(filename,'(i5,''.'',i4,''.'',a4,''.'',a3,''.'',i2,''.eq'')')
     1 nshot,nint(tshot*1000),expnam,dianam,nedit
      do i=1,10
	if(filename(i:i).eq.' ') filename(i:i)='0'
      enddo
      do i=21,22
	if(filename(i:i).eq.' ') filename(i:i)='0'
      enddo
      do i=11,lnblnk(filename)
	if(filename(i:i).eq.' ') filename(i:i)='_'
      enddo
      
      call kkEQintS(ierr,30)
      call kkEQPFM(ierr,expnam,dianam,nshot,nedit,tshot,mdim+1,
     1 m,n,ri,zj,pfm)
      write(*,*) 'kkEQPFM: ierr',ierr
      write(*,*) 'm,n=',m,n
      lpfx=lpfxdim
      call kkEQpfx(iERR,expnam,dianam,nSHOT,nEDIT,tSHOT,
     1 LPFx,PFxx,RPFx,zPFx)
      write(*,*) 'kkEQpfx: ierr',ierr
      rin(1)=rpfx(0)
      zin(1)=zpfx(0)
      write(*,*) 'rpfx',rpfx
      write(*,*) 'zpfx',zpfx
      call kkrzBrzt (iERR,expnam,dianam,nSHOT,nEDIT,tSHOT,
     1 rin,zin,lin,Br,Bz,Bt,fPF,fJp)
      write(*,*) 'kkrzBrzt: ierr',ierr

      rcntc=rin(1)
      btorc=bt(1)
      psilim=pfxx(1)/(2.0*3.14159 26535 89793 23846)
      do i=0,m
	do j=0,n
	  pfmd(i+1,j+1)=pfm(i,j)/(2.0*3.14159 26535 89793 23846)-psilim
	enddo
      enddo
      do i=0,m
	gpr(i+1)=ri(i)
      enddo
      ngr=m+1
      do j=0,n
	gpz(j+1)=zj(j)
      enddo
      ngz=n+1

      print *,'psilim = ',psilim
      print *,'rcntc = ',rcntc
      print *,'btorc = ',btorc
      open(2,file=filename)
      psilim=0
      write(*,*) 'ngr,ngz=',ngr,ngz
      call wreqdg(2,ngpr,ngpz,iret,ngr,ngz,psilim,
     1 btorc,rcntc,gpr,gpz,pfmd)
      if(iret.ne.0) then
          print *,'==== shotfile2dg: error in wreqdg. iret = ',iret
      end if
c
      end
c
      integer function lnblnk(string)
c
c returns the position of the last non-blank character in "string"
c
      character*(*) string
      integer i
      do i=len(string),1,-1
        if(string(i:i).ne.' ') then
          lnblnk=i
          return
        endif
      enddo
      lnblnk=0
      return
      end
